### DISCUSSION TITLE
Zephyrus G14 2020 Ryzen 9 4900HS

### DISCUSSION THREAD

**Discussion by facerollonkeyboard** [PIut02/ROG-Zephyrus-G14-GA401IV-Hackintosh#35] [Q&A] [answered]
I have a g14 2020 with Ryzen 9 4900hs, since it has the same number of cores I directly copied the "2020" folder and downloaded the latest MacOS Sanoma with OpenCore v0.9.9, changed the bios settings using "UMAF". 

As soon as I boot from the USB, I get to the boot selection screen but then, after selecting the "USB", the screen goes black and nothing happens.

I'm sorry, I'm new to Hackintosh, can you please help?

**Comment by oceanondawave** [comment, marked as answer]
> I have a g14 2020 with Ryzen 9 4900hs, since it has the same number of cores I directly copied the "2020" folder and downloaded the latest MacOS Sanoma with OpenCore v0.9.9, changed the bios settings using "UMAF".
> 
> As soon as I boot from the USB, I get to the boot selection screen but then, after selecting the "USB", the screen goes black and nothing happens.
> 
> I'm sorry, I'm new to Hackintosh, can you please help?

Use this [OCAuxiliaryTools](https://github.com/ic005k/OCAuxiliaryTools) to easily edit your config file, google yourself to find out how to use it. Go to your config file > **Misc** > **Boot** > untick **HideAuxiliary**. Save it and reboot, then in the OpenCore Boot Menu you will see an option to boot into the macOS Installer (not the USB one, it has the cog icon if I'm correct).

**Comment by facerollonkeyboard** [comment]
Thank you so much, this worked!
I had to reinstall MacOS 4 times before it worked.

**Reply by oceanondawave** [reply]
Glad it helped!

**Reply by fss-hacks** [reply]
I'm a pretty advanced user that's written 2 different EFIs for very different Intel systems.
This machine (r9 4800hs) is my primary laptop. There are only 3 other mentions (and this is one of them) of this setup working at all on the internet, and at this point I am pretty convinced that reports of any EFI working with this processor are fake as the users refuse to share their EFI publicly even when asked directly.
There is almost no development directly referencing the 4900.
I have tried both the 2020 and 2021 version of this EFI with MacBookPro16,3 SMBIOS and HideAuxiliary unchecked and all UMAF changes made and the installer never makes it past the Apple logo (no progress bar, just reboots).
I cant tell if the issue happening is due to NootedRed (which shouldn't be enabled in the config.plist or even be in the kexts folder when the EFI is distributed because of installer issues) or because of the processor being divergent from the original one the EFI is intended for.
If there is some specific change needed for R9 or any insight you can provide, it would enable me to update the EFI and create a real working example.
Thank you.

**Reply by PIut02** [reply]
> If there is some specific change needed for R9 or any insight you can provide, it would enable me to update the EFI and create a real working example.

First of all from what I understand, 4900hs and 4800hs are essentially the same processor, and I'm leaning towards this as a custom issue.
To get close to this issue, you have to add -v to boot-args to see where the problem is, and you can open a new issue and give a screenshot.

**Reply by fss-hacks** [reply]
I am not even getting to the verbose screen, I’m getting an Apple logo at all times then a restart. 
Im also leaning to there being a config issue going on here, can you give a list of the exact BIOS settings you use?
UMAF has been run and all of that work is done. 
I also need to know if NootedRed should be included or deleted from the EFI during install. 

I was able to get ahold of one of those 3 people and get the BIOS that they used and can provide it to you to look at if you’re interested. It also didn’t work for me. 

Thanks so much for the prompt responses. If I can get this working I will GitHub it like I did with the other EFIs I’ve written and credit you.

**Reply by PIut02** [reply]
Under normal circumstances, you don't need NootedRed to enter and install macOS normally, and I wrote down the modification items in the readme in the BIOS, and the others are the default settings
Maybe you should try reflashing the bios, people who have had a similar issue before and they solved the issue by reflashing the bios.

**Comment by fss-hacks** [comment]
I agree with that too; the issue is to which BIOS version?
This is a GA401IV; I've read that people seem to need BIOS version 0312 to make hackintoshing work but I don't believe that BIOS was ever issued for this machine.
What I can access are versions 218 to 222.
My idea is to match the date of the GA401IV BIOS version to the rough time period 0312 was released and choose the BIOS that comes from that period.
It's strange that the versions for this numerically are so divergent from the other GA401s.

------ Original Message ------

**Reply by PIut02** [reply]
In fact I have used BIOS from 218 to 222 and it works fine.
And 312 I guess this should be used by GA402, which is G14 2022.

**Comment by fss-hacks** [comment]
This is so strange. I downgraded to 220 and still get the empty progress bar.
I see others encountering this when the boot arg and 4G encoding are both enabled at the same time but that’s not the case here.
Can I send you my EFI?

**Comment by fss-hacks** [comment]
Okay now we’re getting somewhere. 
Verbose booted and there’s this kernel panic
![IMG_2470](https://github.com/user-attachments/assets/0b9c4f2f-e67b-4b89-ae41-1a4e492ea17c)

**Comment by fss-hacks** [comment]
Disabled CPUTSCSync and installer booted. 
Had to use an external mouse but surprisingly the Wi-Fi is working for the install.

**Comment by fss-hacks** [comment]
Booted and installed now but the trackpad doesn’t work at all. I know the trackpad is the hardest thing to fix, do you have any idea why it wouldn’t be working?
I did have to disable CPUTSCSync to run the OS, is that connected?

**Reply by PIut02** [reply]
I didn't know much more about the trackpad because I just loaded it and it worked.
Regarding CPUTSCSync, I think maybe an alternative to it should be used [ForgedInvariant](https://github.com/ChefKissInc/ForgedInvariant).
